src/gui/owncloudgui.cpp

```diff
@@ -164,7 +164,13 @@ void ownCloudGui::slotOpenMainDialog()
 
 void ownCloudGui::slotTrayClicked(QSystemTrayIcon::ActivationReason reason)
 {
-    if (reason == QSystemTrayIcon::Trigger || reason == QSystemTrayIcon::Context) {
+    bool macos = false;
+#ifdef Q_OS_MAC
+    macos = true;
+#endif
+
+    if(reason == QSystemTrayIcon::Trigger ||
+            (macos && reason == QSystemTrayIcon::Context)){
```

Comment:
Not a huge fan of that construct, I think I'd prefer either something like:
```cpp
#ifdef Q_OS_MAC
    if (reason == QSystemTrayIcon::Trigger || reason == QSystemTryIcon::Context) {
#else
    if (reason == QSystemTrayIcon::Trigger) {
#endif
```
or:
```cpp
#ifdef Q_OS_MAC
    constexpr auto macos = true;
#else
    constexpr auto macos = false;
#endif

    if (reason == QSystemTrayIcon::Trigger || (macos && reason == QSystemTryIcon::Context)) {
```

Also somehow you changed the coding style of the if (space got removed before the opening parenthesis and after the closing one).

Reply:
Oh BTW... I hoped we didn't even need to do this... out of curiosity, what happens if you unconditionally leave Context in the if? The context menu doesn't show up anymore on Linux and Windows?

Reply:
> Oh BTW... I hoped we didn't even need to do this... out of curiosity, what happens if you unconditionally leave Context in the if? The context menu doesn't show up anymore on Linux and Windows?

You are right I don't need to check for the context click.

Reply:
After you add your new account the menu is displayed in a weird position (more to the left of the systray away from the nc icon instead of near the nc systray icon)

Reply:
Sorry, I think I have been unclear earlier.

What I meant was: I wonder if we even need to change *anything* at all in `slotTrayClicked`!

Indeed, if we're on Mac we test for the `QSystemTrayIcon::Context` reason and we didn't call `setContextMenu` so we'll display the main dialog in both left and right click. If we're not on Mac, we called `setContextMenu`... then I wonder if we get `QSystemTrayIcon::Context` delivered at all. If not, then we don't need to change `slotTrayClicked` otherwise... well, we'll need to do something in `slotTrayClicked` indeed.

Reply:
Removed it and it worked too.